src/slurm_plugin/clustermgtd.py

```diff
@@ -1123,11 +1123,12 @@ def _find_bootstrap_failure_nodes(slurm_nodes):
 
     @staticmethod
     def _find_active_nodes(partitions_name_map):
-        active_nodes = []
+        active_nodes = set()
         for partition in partitions_name_map.values():
             if partition.state != "INACTIVE":
-                active_nodes += partition.slurm_nodes
-        return active_nodes
+                active_nodes |= set(partition.slurm_nodes)
+        # TODO: Returning the set breaks some unit tests (I believe the unit test should be revised though).
+        return sorted(list(active_nodes), key=str)
```

Comment:
what about leaving `active_nodes = []` and at the end returning
```
return list(dict.fromkeys(active_nodes))
```
this should preserve the original order

Reply:
Implemented. Just one doubt: does `list(dict)` always return the list of keys? Is it guaranteed in Python? Should we use `list(dict.fromkeys(active_nodes).keys())` to make it explicit?

Reply:
it's in official doc https://docs.python.org/3/tutorial/datastructures.html#dictionaries
> Performing list(d) on a dictionary returns a list of all the keys used in the dictionary, in insertion order